frontend/public/app.js

```diff
@@ -56,22 +57,28 @@ class Application {
             if (event.target.status >= 200 && event.target.status < 300) {
                 this.fetchDataPeriodically();
             }
+            this.showNotificationByResponseStatus(event.target.status);
         });
 
         let data = {
             text: form.text.value,
-            stock: form.stock.value,
+            // stock: form.stock.value
             uppercase: false,
             reverse: false,
         };
 
-        let json = JSON.stringify(data);
+        if (data.text.trim().length === 0) {
+            this.setAlertMessageContent("Please enter a fruit", "fa fa-exclamation-circle", "pf-c-alert pf-m-danger pf-m-inline");
+            this.showAlertForDuration(3000);
```

Comment:
What's the reasoning for splitting this into two steps? Would there be an issue with a `showAlert(message, options)` type function? Where options could contain the timeout and class info?

Reply:
No design reason i think, was more so how I wrote out the functions, so yeah the whole showAlert can be done as suggested without any issue